exonum/src/blockchain/api_sender.rs

```diff
@@ -45,12 +46,14 @@ impl<T: Send + 'static> ApiSender<T> {
     /// # Return value
     ///
     /// The failure means that the node is being shut down.
-    pub fn send_message(&self, message: T) -> impl Future<Item = (), Error = SendError> {
+    pub async fn send_message(&self, message: T) -> Result<(), SendError> {
```

Comment:
Can be considered later: AFAIK, there is a subtle difference between

```rust
pub async fn send_message(&self, ...)
```

and

```rust
pub fn send_message(&self, ...) -> impl Future<Output = _>
```

The future returned by the first method has a lifetime dependent on `&self`, while the second method will return a future with a `'static` lifetime. Hence, if we reasonably expect that the users will expect a static-lifetime future, then we should use an `impl Future` approach.